confdgnmi/src/confd_gnmi_demo_adapter.py

```diff
@@ -9,18 +16,56 @@ class GnmiDemoServerAdapter(GnmiServerAdapter):
     # map with XPath, value - both strings
     demo_db = {}
     demo_state_db = {}
+    # we use same lock for demo_db and demo_state_db
+    db_lock = threading.Lock()
     num_of_ifs = 10
     _instance: GnmiServerAdapter = None
+    # config map of config elements
+    # "changes" - array of changes contains  (path, val)  tuples or "break"
+    #             or "send" string
+    # "changes_idx" - current index of processing of "changes" array
+    config = {}
 
     capability_list = [
-        dict(name="tailf-common", organization="", version="2020-06-25"),
+        dict(name="tailf-aaa", organization="", version="2018-09-12"),
         dict(name="ietf-inet-types", organization="", version="2013-07-15"),
         dict(name="ietf-interfaces", organization="", version="2014-05-08"),
     ]
 
     def __init__(self):
         self._fill_demo_db()
 
+    @staticmethod
+    def load_config_string(xml_cfg):
+        """
+        Load config from string
+        """
+        log.debug("==> cfg=%s", xml_cfg)
+        root = ET.fromstring(xml_cfg)
+        log.info("root=%s", root)
+        assert root.tag == "demo"
+        changes = root.findall("./subscription/STREAM/changes/element")
+        log.debug("changes=%s", changes)
+        if len(changes):
+            if "changes" not in GnmiDemoServerAdapter.config:
+                GnmiDemoServerAdapter.config["changes"] = []
+                GnmiDemoServerAdapter.config["changes_idx"] = 0
+            for el in changes:
+                log.debug("len(el)=%s", len(el))
+                if len(el):
+                    if len(el) == 2:
+                        path = el[0]
+                        val = el[1]
```

Comment:
C syntax, should be:

```python
(path, val) = el
```

Reply:
Can you assign Element object to tuple?

Reply:
changed to `(path, val) = (el[0], el[1])`